cfme/modeling/base.py

```diff
@@ -200,6 +200,21 @@ def exists(self):
         else:
             return True
 
+    def delete_if_exists(self, *args, **kwargs):
```

Comment:
Q: why `*args` and `**kwargs`?

Reply:
In case a unique entity wants to implement this method separately, something running against many entities can still make the same function call, and pass extra arguments where needed. The core implementation will just ignore the extra args.